File: lib/data/models/notifikasi/notifikasi_model.dart
import 'package:equatable/equatable.dart';

class NotifikasiModel extends Equatable {
  final int id;
  final String judul;
  final String pesan;
  final String tipe; // success, info, warning, error
  final bool isRead;
  final String createdAt;

  const NotifikasiModel({
    required this.id,
    required this.judul,
    required this.pesan,
    required this.tipe,
    required this.isRead,
    required this.createdAt,
  });

  factory NotifikasiModel.fromJson(Map<String, dynamic> json) {
    return NotifikasiModel(
      id: json['id'] ?? 0,
      judul: json['judul'] ?? '',
      pesan: json['pesan'] ?? '',
      tipe: json['tipe'] ?? 'info',
      isRead: json['is_read'] == true || json['is_read'] == 1,
      createdAt: json['created_at'] ?? '',
    );
  }

  @override
  List<Object?> get props => [id, judul, pesan, tipe, isRead, createdAt];
}

File: src/models/notifikasimodel.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');
const Pengguna = require('./penggunamodel');

const Notifikasi = sequelize.define('Notifikasi', {
    id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true
    },
    pengguna_id: {
        type: DataTypes.INTEGER,
        allowNull: false,
        references: {
            model: Pengguna,
            key: 'id'
        }
    },
    judul: {
        type: DataTypes.STRING,
        allowNull: false
    },
    pesan: {
        type: DataTypes.TEXT,
        allowNull: false
    },
    tipe: {
        type: DataTypes.ENUM('success', 'info', 'warning', 'error'),
        defaultValue: 'info'
    },
    is_read: {
        type: DataTypes.BOOLEAN,
        defaultValue: false
    }
}, {
    tableName: 'notifikasi',
    timestamps: true,
    createdAt: 'created_at',
    updatedAt: 'updated_at'
});

// Associations
Pengguna.hasMany(Notifikasi, { foreignKey: 'pengguna_id', onDelete: 'CASCADE' });
Notifikasi.belongsTo(Pengguna, { foreignKey: 'pengguna_id' });

module.exports = Notifikasi;
